### PR TITLE
[omicron-dev] split out components into their own binaries

### PR DESCRIPTION
**Pull request by sunshowers**
I've found myself wanting to run things like MGS via omicron-dev for a while,
but that process requiring all of Nexus to be built has always made it a bit
daunting. Split all the components out into their own binaries to ensure that
the minimum set of dependencies gets built in each case.

There are now separate binaries, all called `-dev` for consistency:

* `cert-dev`: operate on certificates
* `ch-dev`: ClickHouse
* `db-dev`: CockroachDB
* `mgs-dev`: MGS
* `omicron-dev`: just Nexus

There are some workflow changes here:

* `omicron-dev cert-create` -> `cert-dev create`
* `omicron-dev ch-run` -> `ch-dev run`
* `omicron-dev db-run` -> `db-dev run`
* `omicron-dev db-populate` -> `db-dev populate`
* `omicron-dev db-wipe` -> `db-dev wipe`
* `omicron-dev mgs-run` -> `mgs-dev run`

There are also `xtask` aliases for these commands.

But other than that this is just code movement -- there are no underlying changes. (I'll send out an email to all@ about this change.)

An earlier iteration (#6170) attempted to use feature sets to manage this, but
there were legitimate concerns raised there about how
`target/debug/omicron-dev` would have different subcommands based on the last
time you built it. I tried a few alternatives but ultimately like this one the
most.

In the future I'll also add a way to run wicketd via a `wicketd-dev` binary,
and maybe one for sled-agent-sim as well.

### CODE REVIEW

File: dev-tools/omicron-dev/Cargo.toml [hunk 1]
```diff
@@ -12,40 +12,25 @@ omicron-rpaths.workspace = true
...
 expectorate.workspace = true
-libc.workspace = true
+omicron-dev-lib.workspace = true
 omicron-test-utils.workspace = true
-openssl.workspace = true
 oxide-client.workspace = true
 subprocess.workspace = true
-
-# Disable doc builds by default for our binaries to work around issue
-# rust-lang/cargo#8373.  These docs would not be very useful anyway.
```
Comment (sunshowers) [inline dev-tools/omicron-dev/Cargo.toml:48]: This issue was fixed upstream, so I removed this.

Comment (davepacheco) [reply dev-tools/omicron-dev/Cargo.toml:48]: FYI, I think this appears in a bunch of other Cargo.toml files as well.  (Not saying you should fix those here.)

File: docs/how-to-run-simulated.adoc [hunk 2]
```diff
@@ -157,7 +157,7 @@ Note that as the output indicates, this cluster will be available to anybody tha
 +
 [source,text]
 ----
-$ cargo run --bin omicron-dev -- ch-run
+$ cargo run --bin ch-dev run
```
Comment (davepacheco) [inline docs/how-to-run-simulated.adoc:160]: I noticed you dropped a bunch of the `--` arguments here.  Was that intentional?

**Review by davepacheco — APPROVED** [review]
Thanks!  Since you said code was just moved around, I focused here on the doc changes and the changes outside dev-tools.  I didn't look too hard at the stuff inside dev-tools.  I did check that we still have tests for both omicron-dev and db-dev that ^C cleans things up. 👍

File: docs/how-to-run-simulated.adoc (hunk 2 above)
Comment (sunshowers) [reply docs/how-to-run-simulated.adoc]: Ah honestly it doesn't matter, as long as the first argument doesn't start with double-dashes. But it is confusing.

Based on discussion in #oxide-control-plane I added `xtask` aliases which make this problem moot. (Note that it's still possible to run the commands separately, and the big concern we had with a unified binary behind xtask was that omicron-dev would have different sets of arguments based on the last xtask command run. That's not a concern any more.)

**Pull request merged**